File: client/lib/models/paper.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'paper.freezed.dart';
part 'paper.g.dart';

@freezed
class Paper with _$Paper {
  const Paper._();

  const factory Paper({
    required String id,
    required String title,
    required String summary,
    required String keyContribution,
    required String whyItMatters,
    required List<String> authors,
    required String source,
    required String url,
    required String publishedDate,
    required String category,
    required int importanceScore,
    @Default(false) bool isRead,
  }) = _Paper;

  factory Paper.fromJson(Map<String, dynamic> json) => _$PaperFromJson(json);

  String get authorsPreview {
    if (authors.isEmpty) return 'Unknown';
    if (authors.length == 1) return authors.first;
    if (authors.length == 2) return '${authors[0]} & ${authors[1]}';
    return '${authors[0]} et al.';
  }

  String get formattedDate {
    try {
      final dt = DateTime.parse(publishedDate);
      const months = [
        'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun',
        'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'
      ];
      return '${months[dt.month - 1]} ${dt.day}, ${dt.year}';
    } catch (_) {
      return publishedDate;
    }
  }
}

File: server/app/models/paper.py
import uuid
from datetime import datetime, date
from sqlalchemy import String, Text, Integer, DateTime, Date, JSON, Uuid
from sqlalchemy.orm import Mapped, mapped_column
from app.core.database import Base


class Paper(Base):
    __tablename__ = "papers"

    id: Mapped[uuid.UUID] = mapped_column(Uuid, primary_key=True, default=uuid.uuid4)
    title: Mapped[str] = mapped_column(Text)
    summary: Mapped[str] = mapped_column(Text)
    key_contribution: Mapped[str] = mapped_column(Text)
    why_it_matters: Mapped[str] = mapped_column(Text)
    authors: Mapped[list] = mapped_column(JSON)
    source: Mapped[str] = mapped_column(Text)
    url: Mapped[str] = mapped_column(Text, unique=True)
    published_date: Mapped[date] = mapped_column(Date)
    category: Mapped[str] = mapped_column(String(100))
    importance_score: Mapped[int] = mapped_column(Integer)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)
